### PR TITLE
rusk-abi: Add BLS signature verification

### LINKED ISSUE #475: Add BLS signature verification to rusk-abi
**Describe what you want implemented** 
Implement `dusk-bls12_381-sign` verification as part of `rusk-abi` module.

**Describe "Why" this is needed**
This is needed to verify the signature produced by the BLS keys owned by the node for stake.

**Describe alternatives you've considered**
The node might use jubjub keys. However, that is a huge change in the consensus with high complexity and potential performance impact.

### PR DESCRIPTION
**Pull request by vlopes11**
Resolves #475

### CODE REVIEW

**Review by ureeves — APPROVED** [review]
LGTM

File: rusk-abi/src/module/host.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ use alloc::vec::Vec;
 use canonical::{Canon, CanonError, Source};
 use dusk_abi::{ContractId, HostModule, Query, ReturnValue};
 use dusk_bls12_381::BlsScalar;
+use dusk_bls12_381_sign::{Signature as BlsSignature, APK as BlsPublicKey};
```
Comment (herr-seppia) [inline rusk-abi/src/module/host.rs:13]: I would avoid to alias APK as PublicKey, it can lead to misunderstanding IMO

File: rusk-abi/tests/lib.rs [hunk 2]
```diff
@@ -198,6 +199,61 @@ mod module_tests {
...
+
+        let contract_id = network.deploy(contract).unwrap();
+
+        let mut gas = GasMeter::with_limit(1_000_000_000);
+
+        let message = b"some-message".to_vec();
+
+        let sk = BlsSecretKey::random(&mut rand_core::OsRng);
+        let pk = (&sk).into();
+        let apk = BlsPublicKey::from(&sk);
```
Comment (herr-seppia) [inline rusk-abi/tests/lib.rs:224]: Any particular reason in create the APK from SecretKey instead of from PublicKey?

**Review by herr-seppia — COMMENTED** [review]
LGTM overall

Just some nits/clarifications in the comment

File: rusk-abi/tests/lib.rs (hunk 2 above)
Comment (herr-seppia) [reply rusk-abi/tests/lib.rs:224]: Ok, I didn't notice this #52
My fault, I flag this comment as resolved

File: rusk-abi/src/module/host.rs (hunk 1 above)
Comment (vlopes11) [reply rusk-abi/src/module/host.rs]: Renamed to `AggregatedBlsPublicKey`

**Pull request merged**